src/parse.rs

```diff
@@ -222,6 +223,26 @@ pub fn parse_mailbox(mut lines: &[u8]) -> Result<Mailbox> {
     }
 }
 
+pub fn parse_search_ids(lines: &[u8]) -> Result<Vec<u32>> {
+    match str::from_utf8(lines) {
+        Ok(resp) => {
+            let re = Regex::new(r"^(?i)\* SEARCH((?:\s+[0-9]+)*)\s*$").unwrap();
```

Comment:
I think this'll fail on multi-line responses. It also requires compiling a regex, which is a little sad. In `buzz`, I do this with `split_whitespace` instead: https://github.com/jonhoo/buzz/blob/b2a94bc62b45556ff630bf3f1b59a65af56aabe1/src/main.rs#L111, but the "right" way to do it is using `imap-proto`. This is partially so that we don't have to duplicate parsing code between the different Rust IMAP clients, bu partially to deal with the fact that the server may send unilateral responses at any time. See https://github.com/mattnenterprise/rust-imap/commit/72925cf1d81e0812375ce7a09ccb99b401078298. The other parsers work around this by using `parse_many`, which calls `imap_proto::parse_response` on the returned string. In the case of `SEARCH` and `UID SEARCH`, the response type should be [`Response::IDs`](https://docs.rs/imap-proto/0.4/imap_proto/types/enum.Response.html) I think, which will already hold the parsed numeric IDs. I think using [`parse_capabilities`](https://github.com/mattnenterprise/rust-imap/blob/1a62f1b24b2c00e34c7747fff1e948383e1924ee/src/parse.rs#L129) for inspiration will get you most of the way there!

Reply:
Yeah, this was where I knew I should have used a different approach.

I'll take a look at how to reimplement this. Thanks for the pointers.

Reply:
I think I managed to do it pretty easily. As you said it was almost a copy-paste of `parse_capabilities`. The tests run fine.